utils/docker/Dockerfile.centos.8

```diff
@@ -12,29 +12,78 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories that can be mirrored.
+# The Docker image starts out with the DISTRO provided GPG keys
+# on the disk, but not installed.  These must be isntalled before
+# adding any repo that needs GPG keys but does not provide meta
+# data for installing those keys.  Then the epel-release
+# package must be installed and the GPG key it provides must be
+# installed.
 ARG REPO_URL
-# Need to clean this up later
-# This is largely the same as the centos.7 code, however this sets priority for
-# the new repo, and leaves the default ones enabled as there is some metadata
-# that is not available locally.
-RUN if [ -n "$REPO_URL" ]; then						                           \
-      version=$(sed -e 's/.* \([0-9\.]*\)\.[0-9]*.*/\1/' /etc/centos-release); \
-      MY_REPO="${REPO_URL}repository/centos-${version}-x86_64-group/";         \
-      MY_NAME="${MY_REPO#*//}";						                           \
-      MY_NAME="${MY_NAME//\//_}";					                           \
+ARG REPO_DISTRO=""
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/*;                          \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
       echo -e "[${MY_NAME}]\n\
 name=created from ${MY_REPO}\n\
 baseurl=${MY_REPO}\n\
 enabled=1\n\
-priority=1\n\
-gpgcheck=False\n" >> /etc/yum.repos.d/local-centos-group.repo;                 \
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      dnf -y install --disablerepo extras --disablerepo baseos  \
+          epel-release-8-8.el8 dnf-plugins-core dnf-utils;      \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-8;         \
+      dnf config-manager --assumeyes --quiet --disable          \
+          baseos powertools epel epel-modular extras;           \
+      sed "s/^mirrorlist_expire=0*/mirrorlist_expire=99999999/" \
+          -i /etc/dnf/dnf.conf;                                 \
+      dnf -y upgrade epel-release;                              \
+    else                                                        \
+      dnf -y install epel-release dnf-plugins-core &&           \
+      dnf config-manager --set-enabled powertools;              \
```

Comment:
It seems that this `else` block is what is missing from the EL7 Dockerfile as I previously commented.

Reply:
CentOS 8 needed the else block, because enabling a repo with dnf enables the upstream repository so must only be done if the local repository is not done.

CentOS 7 re-install of epel-release just reports nothing to do for it.